src/v/kafka/server/handlers/configs/config_utils.h

```diff
@@ -320,39 +326,68 @@ struct replication_factor_must_be_positive {
 };
 
 struct replication_factor_must_be_odd {
-    std::optional<ss::sstring>
-    operator()(const ss::sstring&, const cluster::replication_factor& value) {
+    std::optional<ss::sstring> operator()(
+      model::topic_namespace_view,
+      const ss::sstring&,
+      const cluster::replication_factor& value) {
         if (value % 2 == 0) {
             return fmt::format("replication factor {} must be odd", value);
         }
         return std::nullopt;
     }
 };
 
+struct replication_factor_must_be_greater_or_equal_to_minimum {
+    std::optional<ss::sstring> operator()(
+      model::topic_namespace_view tns,
+      const ss::sstring&,
+      const cluster::replication_factor& value) {
+        if (model::is_user_topic(tns)) {
+            if (auto& min_val
+                = config::shard_local_cfg().minimum_topic_replication.value();
```

Comment:
Did we determine this can't race? Or what was the verdict there?

Reply:
I believe the issue was with the cluster configurations not the configs with alter configs (or am I not remembering something correctly?)

Reply:
If you're referring to the cluster configuration race condition, I went ahead and removed the linking between the two configs (which reminds me I need to update the cover sheet for this PR)